Data/Vector/Storable/Mutable.hs

```diff
@@ -371,15 +371,49 @@ clone = G.clone
 -- Growing
 -- -------
 
--- | Grow a vector by the given number of elements. The number must be
--- positive.
+-- | Grow a storable vector by the given number of elements. The number must be
+-- non-negative. Same semantics as in `G.grow` for generic vector.
+--
+-- ====__Examples__
+--
+-- >>> import qualified Data.Vector.Storable as VS
+-- >>> import qualified Data.Vector.Storable.Mutable as MVS
+-- >>> mv <- VS.thaw $ VS.fromList ([10, 20, 30] :: [Int])
+-- >>> mv' <- MVS.grow mv 2
+--
+-- Extra elements at the end of the newly allocated vector are initialized to
+-- default value, which for storable vector will always be 0. However, if
```

Comment:
Not all `Storable` types are numerics (`Bool`, `Char`, `()`, etc.). Same for `Primitive` above.

Reply:
Sure, but it doesn't change the fact that memory will contain zeros.

Reply:
If you have better wording for this, I am all ears.

Reply:
How about "Extra memory is initialized with zero bytes" or similar, without refering to elements? There are quite non-trivial instances of `Storable` in the wild, for which zeros does not mean anything.

Reply:
@Bodigrim I did rephrase it a bit, hopefully it makes things more clear. This sort of explanation ideally would be explained in haddock for `new`, but one thing at a time.